feat: add standalone guide on XLA support.

sayakpaul opened the pull request:
# What does this PR do?

We have had XLA support for our TF generation models (GPT2, Whisper, for example) for a while. This PR adds a standalone guide in the doc to discuss it. 

Cc: @Rocketknight1 @amyeroberts @ydshieh

sayakpaul committed:
feat: add standalone guide on XLA support.

Co-authored-by: Joao Gante <joaofranciscocardosogante@gmail.com>

ydshieh commented:
@sayakpaul 

It seems there is an issue with your CircleCI permissions.
Could you try refreshing your permissions as shown [here](https://support.circleci.com/hc/en-us/articles/360048210711-How-to-Refresh-User-Permissions-)?

sayakpaul commented:
@ydshieh I don't have CircleCI signed in nor have I installed it here: https://github.com/settings/applications.

ydshieh commented:
Could you login CircleCI with your GitHub account, and follow `huggingface/transformers`?

See https://circleci.com/docs/projects/

(but it's kinda strange - you have opened a lot of PRs before, so not sure why we have this issue now)

sayakpaul commented:
Should be all good now I guess:

<img width="896" alt="image" src="https://user-images.githubusercontent.com/22957388/212717447-19e1b042-d39f-44f0-bf8c-faef7b97c2ea.png">

sayakpaul committed:
Empty commit to trigger CI

HuggingFaceDocBuilderDev commented:
The docs for this PR live [here](https://moon-ci-docs.huggingface.co/docs/transformers/pr_21141). All of your documentation changes will be reflected on that endpoint.

gante reviewed (approved):
(As pre-approved in the google docs draft :) )

sgugger commented on docs/source/en/tf_xla.mdx:
````diff
@@ -0,0 +1,162 @@
+_ = model(random_inputs)
+```
+
+In order to run the forward pass with an XLA-compiled function, we’d need to:
````
```suggestion
In order to run the forward pass with an XLA-compiled function, we’d need to do:
```

sgugger commented on docs/source/en/tf_xla.mdx:
```diff
@@ -0,0 +1,162 @@
+
+## Running a TF text generation model with XLA from 🤗 Transformers
+
+To enable XLA-accelerated generation within 🤗 Transformers, you need to have the correct version of `transformers` installed. You can install it by running:
```
```suggestion
To enable XLA-accelerated generation within 🤗 Transformers, you need to have a recent version of `transformers` installed. You can install it by running:
```
(the suggests there is only one)

sgugger commented on docs/source/en/tf_xla.mdx:
````diff
@@ -0,0 +1,162 @@
+To enable XLA-accelerated generation within 🤗 Transformers, you need to have the correct version of `transformers` installed. You can install it by running:
+
+```bash
+pip install transformers>=4.21.0
````
```suggestion
pip install transformers --upgrade
```
We want users to be on the latest version.

sgugger commented on docs/source/en/tf_xla.mdx:
````diff
@@ -0,0 +1,162 @@
+```bash
+pip install transformers>=4.21.0
+```
+
````
Maybe add a code sample that displays the version and instruct the user to check it is greater than 4.21.0?

sgugger reviewed (approved):
Thanks for this new guide. You should make it an auto-generated notebook by adding `[[open-in-colab]]` at the top.

sayakpaul committed:
Apply suggestions from code review

Co-authored-by: Sylvain Gugger <35901082+sgugger@users.noreply.github.com>

sayakpaul committed:
address PR comments.

sayakpaul commented:
@sgugger addressed your comments including [this one](https://github.com/huggingface/transformers/pull/21141/files#r1072172489). Let me know if I am good to merge given all tests are green.

sgugger commented:
Yes, all good!
